cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -128,13 +128,19 @@ func (c *client) EnsureNamespace(namespaceName string) error {
 		return errors.Wrap(err, "error creating core clientset")
 	}
 
+	namespaces, err := clientset.CoreV1().Namespaces().List(metav1.ListOptions{})
```

Comment:
Any particular reason to use `List` here instead of `Get`?

This is also missing a `if err != nil` check. It might be good to test the error against `apierrors.IsForbidden` to see if the user is also missing permissions to List or Get Namespaces, in which case we can likely emit a warning to the user.